org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/SynchronizeGradleBuildOperation.java

```diff
@@ -154,16 +154,22 @@ public File apply(OmniEclipseProject gradleProject) {
 
             @Override
             public boolean apply(IProject project) {
-                Optional<ProjectConfiguration> projectConfiguration = CorePlugin.projectConfigurationManager().tryReadProjectConfiguration(project);
-                return projectConfiguration.isPresent()
-                        && projectConfiguration.get().toRequestAttributes(ConversionStrategy.MERGE_WORKSPACE_SETTINGS).getProjectDir().equals(SynchronizeGradleBuildOperation.this.build.getProjectDir())
+                BuildConfiguration buildConfiguration = CorePlugin.configurationManager().loadProjectConfiguration(project).getBuildConfiguration();
+                return buildConfiguration.getRootProjectDirectory().equals(SynchronizeGradleBuildOperation.this.buildConfig.getRootProjectDirectory())
                         && (project.getLocation() == null || !gradleProjectDirectories.contains(project.getLocation().toFile()));
             }
         }).toList();
     }
 
     private void synchronizeGradleProjectWithWorkspaceProject(OmniEclipseProject project, SubMonitor progress) throws CoreException {
         progress.setWorkRemaining(1);
+
+        // save the project config before creating/opening workspace project to avoid any premature updates
+        // This ensures that events like `ProjectCreatedEvents` are not fired before the configuration is available
```

Comment:
They aren't fired anyway early, because we run in a transaction. Also, at this point we already have an open workspace project, so this comment is misleading.